web_src/js/utils.ts

```diff
@@ -43,6 +43,16 @@ export function parseIssueNewHref(href: string): IssuePathInfo {
   return {ownerName, repoName, pathType, indexString};
 }
 
+export function parseIssuePageInfo(): IssuePageInfo {
+  const el = document.querySelector('#issue-page-info');
+  const info = {} as IssuePageInfo;
+  info.issueNumber = parseInt(el?.getAttribute('data-issue-index'));
+  info.issueDependencySearchType = el?.getAttribute('data-issue-dependency-search-type') || '';
+  info.repoId = parseInt(el?.getAttribute('data-issue-repo-id'));
+  info.repoLink = el?.getAttribute('data-issue-repo-link') || '';
```

Comment:
```suggestion
  const info: IssuePageInfo = {
    issueNumber: parseInt(el?.getAttribute('data-issue-index'));
    issueDependencySearchType: el?.getAttribute('data-issue-dependency-search-type') || '';
    repoId: parseInt(el?.getAttribute('data-issue-repo-id'));
    repoLink: el?.getAttribute('data-issue-repo-link') || ''; 
  }
```

`as` should generally be avoided.